### ISSUE TITLE
Volume stays in "InUse" state even when no pods are using it

### ISSUE THREAD

**Issue by vitobotta** [LINBIT/linstor-csi#29]
Hi! I am having an issue where if for example I scale a deployment to zero or delete the pod or upgrade the deployment (e.g. changing the image), the volume remains in "InUse" state according to the Linstor CLI, and then the new pod cannot use the volume:

```
Multi-Attach error for volume "pvc-9802166a-b921-11e9-bbca-9600002d2434" Volume is already exclusively attached to one node and can't be attached to another
```

This happens even if the deployment is configured to kill the old pod before starting the new one, so that only one pod is using the volume at any time.

Unless I am missing something I would expect that if for example if the deployment doesn't have any pods, the volume should become "Unused" so that it can be attached/mounted to the new pod when scaling the deployment to 1 or anyway when the new pod is created.

Here's the most recent logs: https://pastebin.com/kAXQQsTF

Thanks!

**Comment by vitobotta** [comment]
Just to add, this is what I see with lsblk when I no longer have pods: 

```
sdb                                                            8:16    0   100G  0 disk
├─vg-lvmthinpool_tmeta                                       253:0     0   100M  0 lvm
│ └─vg-lvmthinpool-tpool                                     253:2     0  99.8G  0 lvm
│   ├─vg-lvmthinpool                                         253:3     0  99.8G  0 lvm
│   └─vg-pvc--9802166a--b921--11e9--bbca--9600002d2434_00000 253:4     0   200G  0 lvm
│     └─drbd1001                                             147:1001  0   200G  0 disk /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-9802166a-b921-11e9-bbca-9600002d2434/globalmount
└─vg-lvmthinpool_tdata                                       253:1     0  99.8G  0 lvm
  └─vg-lvmthinpool-tpool                                     253:2     0  99.8G  0 lvm
    ├─vg-lvmthinpool                                         253:3     0  99.8G  0 lvm
    └─vg-pvc--9802166a--b921--11e9--bbca--9600002d2434_00000 253:4     0   200G  0 lvm
      └─drbd1001                                             147:1001  0   200G  0 disk /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-9802166a-b921-11e9-bbca-9600002d2434/globalmount
```

The volume remains mounted to a path.

**Comment by vitobotta** [comment]
Also for example the volume is currently attached to the node linstor-master1 and a new pod has been scheduled on linstor-master2, so the volume cannot be attached to the latter.

**Comment by deleteriousEffect** [comment]
Can you upgrade the CSI driver to 0.6.4 and try to reproduce? There's a change relevant to mounting/unmounting which should have a positive effect on this issue. I'm not 100% convinced that it will solve it, but it's a step in the right direction.

**Comment by vitobotta** [comment]
I'm already using 0.6.4 :)

**Comment by deleteriousEffect** [comment]
If you're not using the `localStoragePolicy` parameter. You should be able to upgrade to 0.7.0 without an issue.

**Comment by vitobotta** [comment]
I tried to update to 0.7.0 earlier but the images were missing - https://quay.io/repository/linbit/linstor-csi?tab=tags shows 0.6.4 as the latest.

**Comment by vitobotta** [comment]
```
Back-off pulling image "quay.io/linbit/linstor-csi:v0.7.0"
```

**Comment by deleteriousEffect** [comment]
Try it now, it should be up.

**Comment by vitobotta** [comment]
Thanks, trying with 0.7.0 now. Something seems different in that it now if it detaches the volume it seems, but if the pod is rescheduled on a different node when I scale up to 1, then I still get `Volume is already exclusively attached to one node and can't be attached to another` even though the volume is now 'Unused'.

**Comment by vitobotta** [comment]
I restarted the nodes one per time just in case and it seems to be working now! Still testing but it seems that the volume is detached and reattached on different nodes as the pod is rescheduled. Thanks! :)

**Comment by deleteriousEffect** [comment]
Are these volumes that you initially attached to the pods with an older version of the CSI driver? If so, what the cause might be due to eec71a37c508281e117be854db1fe875997dc682. Before then the volumes were essentially mounted twice and unmounted twice. So a volume that was mounted the old way and unmounted the new way would probably produce the behavior your observed.

**Comment by vitobotta** [comment]
Yes, I had a couple volumes attached when I upgraded. Is that why rebooting helped?

**Comment by deleteriousEffect** [comment]
That's right, reboot is the ultimate unmount operation :laughing:

**Comment by vitobotta** [comment]
Cool. I'll close this since the issue seems to be fixed with 0.7.0. Thanks again :)
